### DISCUSSION TITLE
More customization for diffTool

### DISCUSSION THREAD

**Discussion by andrewjmeier** [pointfreeco/swift-snapshot-testing#623] [Ideas]
I'd like to use Imagemagick's `compare` to diff snapshots. It expects the format `compare [fileA] [fileB] -compose src [outputFileName]`. With the way it's currently written, `diffTool` only works for tools that use the format `[tool] [fileA] [fileB]`. How about the option to add an `errorMessageBuilder` block that can construct an error message in whatever format is needed?

**Comment by jlcvp** [comment]
#668 seems to address this

**Reply by andrewjmeier** [reply]
Not quite as flexible as what I proposed but that would work for me.

**Comment by js** [comment]
This is an old discussion, but I added a PR for a diff tool builder closure in #840
